### ISSUE TITLE
Could not autoload puppet/parser/functions/oracle_sso: no such file to load

### ISSUE THREAD

**Issue by squaricdot** [antoineco/aco-oracle_java#45]
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Function Call, Could not autoload puppet/parser/functions/oracle_sso: no such file to load -- /etc/puppetlabs/code/modules/oracle_java/lib/puppet_x/aco/util at /etc/puppetlabs/code/environments/my_dev/modules/oracle_java/manifests/download.pp:33:17

Puppet agent 5.3.3
Puppet server 5.1.4

Fixed it for myself by comparing it to another module (archive)...

The oracle_sso is requiring on puppet_x/aco/util

but it seems it prefers the full name of the ruby module.

**Comment by antoineco** [comment]
This is fixed in the latest release (v2.9.2). Thanks for reporting! 🙌

**Comment by ngeorgieff** [comment]
I'm getting the same error with version 2.9.3
- puppet-enterprise-release-2017.3.0.0-1
- puppet agent 5.3.2-1

```
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Loading facts
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Function Call, Could not autoload puppet/parser/functions/oracle_sso: undefined method `path' for nil:NilClass at /etc/puppetlabs/code/environments/development/modules/oracle_java/manifests/download.pp:33:17 on node
```
